consensus/reactor.go

```diff
@@ -1488,6 +1595,21 @@ type VoteSetBitsMessage struct {
 	Votes   *cmn.BitArray
 }
 
+// ValidateBasic performs basic validation.
+func (m *VoteSetBitsMessage) ValidateBasic() error {
+	if m.Height < 0 {
+		return errors.New("Negative Height")
+	}
+	if m.Round < 0 {
+		return errors.New("Negative Round")
+	}
+	if err := m.BlockID.ValidateBasic(); err != nil {
+		return fmt.Errorf("Wrong BlockID: %v", err)
+	}
+	// NOTE: Votes.Size() can be zero if the node does not have any
```

Comment:
It's a bit hard to read the code since we have these `if nil return nil` bits. Here I think we can return nil https://github.com/tendermint/tendermint/blob/c8cf78039f45a08c7f8fec7fd7acc12d47090832/types/vote_set.go#L345 if we don't have prevotes/precommits for that blockID